include/irr/video/decodePixels.h

```diff
@@ -1283,6 +1286,17 @@ namespace irr { namespace video
                 else s = std::pow((s + 0.055) / 1.055, 2.4);
             }
         }
+		
+        template<>
+        inline void lin2SRGB<double>(double _lin[3])
```

Comment:
@Crisspl  has this very function in `encodePixels.h` I believe?

Reply:
Yes, but `lin2SRGB()` on `encodePixels.h` only works with `double`, and downcasting is ugly. I can move my conversion functions there though.

Reply:
you can make his templated, and slam an assert about the template type (onyl float or double) in there